refactor: include/exclude dev deps in analyzers

DmitriyLewen opened the pull request:
## Description
Move dev(test) dependencies inclusion/exclusion in analyzers.
See #7476 for more details.

## Related issues
- Close #7476 


## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

coheigea commented:
@DmitriyLewen https://github.com/aquasecurity/trivy/commit/2d97700d10665142d2f66d7910202bec82116209 caused a major regression for us, in that it appears --include-dev-deps was never wired through, and maven test dependencies are therefore included by default. Will it be fixed in this PR? IMO it's worthy of a hotfix as we had to downgrade to stop all of our builds breaking.

DmitriyLewen commented:
Hello @coheigea 
yeah, this PR contains fix for that (https://github.com/aquasecurity/trivy/pull/7484/commits/dba9f9f7f03afe6dd3cb111e3a14bcb050233303)

Can you test these changes in your project?

coheigea commented:
Thanks @DmitriyLewen , with the PR it doesn't scan test dependencies by default any more.

DmitriyLewen commented on pkg/fanal/analyzer/language/java/pom/pom.go:
```diff
@@ -21,10 +21,18 @@ func init() {
 	filePath := filepath.Join(input.Dir, input.FilePath)
+
+	// Skip integration test dir if `--include-dev-deps` flag is not present.
+	if isIntegrationTestDir(filePath) && !a.includeDevDeps {
+		return nil, nil
+	}
```
I was thinking about moving this check to the `Required` function.
But it might not be entirely obvious

knqyf263 reviewed (commented):
We need to take the flag into account for cache key calculation.
https://github.com/aquasecurity/trivy/blob/e8c32dedaa58ed459f8d7991ab456d12b71cef6e/pkg/cache/key.go#L27-L35

Also, some more analyzers support development dependencies now.
https://github.com/aquasecurity/trivy/blob/e8c32dedaa58ed459f8d7991ab456d12b71cef6e/pkg/dependency/parser/python/uv/parse.go#L159

DmitriyLewen commented:
> We need to take the flag into account for cache key calculation.

I didn't change the key for the cache - because all analyzers don't work for the image mod.
But since we are now talking about the ability to enable/disable analyzers (#3987) - this will be useful, thanks.
updated in https://github.com/aquasecurity/trivy/pull/7484/commits/18c4b678b9ccbf98307c1575098cea0a9077d0da

> Also, some more analyzers support development dependencies now.

Updated missing analyzers.
